src/components/Comments/CommentBox.jsx

```diff
@@ -0,0 +1,65 @@
+import { useState, useRef } from "react";
+import * as api from "../../api";
+import toast from "react-hot-toast";
+
+const CommentBox = ({ setComments, setCommentCount, reviewId }) => {
+  const [inputText, setInputText] = useState("");
+  const previousComments = useRef([]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setInputText("");
+    setCommentCount((currentCommentCount) => {
+      return currentCommentCount + 1;
+    });
+    setComments((currentComments) => {
+      previousComments.current = currentComments;
+      const currentTime = Date.now();
+      const newComment = {
+        author: "jessjelly",
+        body: inputText,
+        votes: 0,
+        created_at: currentTime,
+        comment_id: currentTime, //make a unique comment_id to satisfy react's unique key on children requirement, server will send the real comment_id when users get the comment for real
+      };
+      return [newComment, ...currentComments];
+    });
+
+    api
+      .postComment({ username: "jessjelly", body: inputText }, reviewId)
+      .catch((error) => {
+        console.error(error);
+        toast.error("Comment posting failed! Please try again later.");
```

Comment:
Really like the conditional rendering here with the notification if the request fails!